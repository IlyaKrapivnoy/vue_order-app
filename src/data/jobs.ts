import Job from '@/types/Job';

const jobs: Job[] = [
  {
    title: 'JavaScript Developer',
    location: 'lisbon, Portugal',
    salary: 50600,
    id: '1',
  },
  {
    title: 'Vue Developer',
    location: 'Porto, Portugal',
    salary: 40200,
    id: '2',
  },
  {
    title: 'React Developer',
    location: 'Berlin, Germany',
    salary: 40500,
    id: '3',
  },
];

export default jobs;
