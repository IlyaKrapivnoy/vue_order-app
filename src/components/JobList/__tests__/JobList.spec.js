import { mount } from '@vue/test-utils';
import JobList from '../JobsList.vue';
import jobs from '@/data/jobs';

describe('JobList.vue', () => {
  it('renders ordered job list', async () => {
    // Mount the component with props
    const wrapper = mount(JobList, {
      props: {
        jobs,
        order: 'title',
      },
    });

    // Wait for Vue to update the component
    await wrapper.vm.$nextTick();

    // Assert that the ordered job list is displayed correctly
    const jobTitles = wrapper.findAll('.job-list h2');
    expect(jobTitles.length).toBe(3);
    expect(jobTitles[0].text()).toBe(
      'JavaScript Developer in Lisbon, Portugal',
    );
    expect(jobTitles[1].text()).toBe('React Developer in Berlin, Germany');
    expect(jobTitles[2].text()).toBe('Vue Developer in Porto, Portugal');

    // Use Jest's toMatchSnapshot to create a snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });
});
