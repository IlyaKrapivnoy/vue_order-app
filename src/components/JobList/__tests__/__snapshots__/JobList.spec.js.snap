// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobList.vue renders ordered job list 1`] = `
<div class="job-list">
  <p>Ordered by title</p>
  <ul>
    <li>
      <h2>JavaScript Developer in Lisbon, Portugal</h2>
      <div class="salary">
        <p>50600 euro</p>
      </div>
      <div class="description">
        <p> Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet aspernatur et hic, magnam minus natus quae? Aspernatur dolor illum, in laudantium libero maiores minima officiis quas quidem quod, veniam voluptate. </p>
      </div>
    </li>
    <li>
      <h2>React Developer in Berlin, Germany</h2>
      <div class="salary">
        <p>40500 euro</p>
      </div>
      <div class="description">
        <p> Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet aspernatur et hic, magnam minus natus quae? Aspernatur dolor illum, in laudantium libero maiores minima officiis quas quidem quod, veniam voluptate. </p>
      </div>
    </li>
    <li>
      <h2>Vue Developer in Porto, Portugal</h2>
      <div class="salary">
        <p>40200 euro</p>
      </div>
      <div class="description">
        <p> Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet aspernatur et hic, magnam minus natus quae? Aspernatur dolor illum, in laudantium libero maiores minima officiis quas quidem quod, veniam voluptate. </p>
      </div>
    </li>
  </ul>
</div>
`;
